common/src/main/java/com/selfdot/cobblemonmegas/common/CobblemonMegas.java

```diff
@@ -105,6 +107,8 @@ private void onServerStarting(MinecraftServer server) {
 
     private Unit onBattleStartedPre(BattleStartedPreEvent event) {
         MegaUtils.deMegaEvolveAllPlayers(event.getBattle());
+        var players = event.getBattle().getPlayers();
+        players.forEach(player -> Cobblemon.playerData.get(player).getKeyItems().add(new Identifier("cobblemon", "key_stone")));
```

Comment:
You may want to add additional handling for this with your keystones, up to your discretion

Reply:
Do you know if there's an API for certain items being key items, or do I need to add and remove "cobblemon:key_stone" when the items are put in/taken from the player's inventory?

Reply:
I don't believe there is API for that yet. Looking at references keystones property shows only an extension function for retrieval.

If you want to add proper keystone logic, the route I would pursue first would be:

- Modify the `BattleStartedPreEvent` handler to add the identifier only if player has item in inventory
- Hook into the `BattleFledEvent` and `BattleVictoryEvent` and then remove identifiers

The one edge case I can think of here would be server crashing during and the player data becoming stale, so you would also want to remove the key in the `BattleStartedPreEvent` event if it's _not_ in the inventory.

Reply:
Actually, I think either adding or removing (if it exists) on `BattleStartedPreEvent` depending on if it's in inventory should cover it? Shouldn't need to complicate it with the end-of-battle logic.

Reply:
Perfect, thanks.